lib/ansiblelint/formatters/__init__.py

```diff
@@ -1,15 +1,17 @@
 """Output formatters."""
 import os
 from pathlib import Path
-from typing import TYPE_CHECKING, Union
+from typing import TYPE_CHECKING, Generic, TypeVar, Union
 
 from ansiblelint.color import Color, colorize
 
 if TYPE_CHECKING:
     from ansiblelint.errors import MatchError
 
+T = TypeVar('T', bound='BaseFormatter')
 
-class BaseFormatter:
+
+class BaseFormatter(Generic[T]):
```

Comment:
Not sure if it's necessary. It looks like it's needed to just use custom type @ `__main__`

Reply:
See my previous comment.